### PR TITLE
Support Polish

### PR DESCRIPTION
**Pull request by oliverkinch**
# Datasets

- Reading comprehension: [PoQuAD](https://huggingface.co/datasets/clarin-pl/poquad) 
- Sentiment classification: [PolEmo2.0](https://huggingface.co/datasets/clarin-pl/polemo2)
- Linguistic acceptability: ScaLA algorithm with [Polish universal dependencies](https://github.com/UniversalDependencies/UD_Polish-PDB)
- Named entity recognition: [KPWR-NER](https://huggingface.co/datasets/clarin-pl/kpwr-ner)
- Summarization: [PSC](https://huggingface.co/datasets/community-datasets/psc)
- Knowledge: [LLMzSzL](https://huggingface.co/datasets/amu-cai/llmzszl-dataset)

Fix https://github.com/EuroEval/EuroEval/issues/704

### CODE REVIEW

**Comment by oliverkinch** [comment]
To run `train = load_dataset("clarin-pl/poquad", split="train")` I needed to downgrade `datasets==3.6.0` (https://github.com/huggingface/datasets/issues/7693#issuecomment-3103380232)

**Comment by oliverkinch** [comment]
Should I also take a look at:

Potentially NLI datasets (would need to be set up): [SICK-pl](https://huggingface.co/datasets/sdadas/sick_pl) and [CDSCorpus](https://zil.ipipan.waw.pl/Scwad/CDSCorpus). Could potentially replace the common-sense reasoning.

(from https://github.com/EuroEval/EuroEval/issues/704#issue-2782621754)

File: docs/datasets/polish.md [hunk 1]
```diff
@@ -72,6 +72,70 @@ $ euroeval --model <model-id> --dataset polemo2
 
 ## Linguistic Acceptability
 
+### ScaLA-Pl
```
Comment (saattrupdan) [inline docs/datasets/polish.md]:
```suggestion
### ScaLA-pl
```

File: docs/datasets/polish.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
+# 🇵🇱 Polish
+
+This is an overview of all the datasets used in the Polish part of EuroEval. The
+datasets are grouped by their task - see the [task overview](/tasks) for more
+information about what these constitute.
+
+## Sentiment Classification
+
+### PolEmo 2.0
```
Comment (saattrupdan) [inline docs/datasets/polish.md]: Ensuring consistency with dataset name in the CLI.

```suggestion
### PolEmo2
```

File: docs/datasets/polish.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+- Label mapping:
+    - `correct` ➡️ `tak`
+    - `incorrect` ➡️ `nie`
+
+You can evaluate this dataset directly as follows:
+
+```bash
+$ euroeval --model <model-id> --dataset scala-pl
+```
+
```
Comment (saattrupdan) [inline docs/datasets/polish.md:206]:
```suggestion


```

File: docs/datasets/polish.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+## Reading Comprehension
+
+### PoQuAD
+
+PoQuAD is a Polish Question Answering dataset with contexts from Polish Wikipedia. It follows the SQuAD format with innovations including lower annotation density, abstractive answers, polar questions, and impossible questions.
```
Comment (saattrupdan) [inline docs/datasets/polish.md:212]: We normally remove the impossible questions - has that been done here?

Also, are the samples human-labelled, automatic, or is a translation?

File: src/euroeval/dataset_configs/polish.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    pretty_name="the Polish question answering dataset PoQuAD",
+    huggingface_id="EuroEval/poquad",
+    task=RC,
+    languages=[PL],
+)
+
+POLEMO2_CONFIG = DatasetConfig(
+    name="polemo2",
+    pretty_name="the Polish sentiment analysis dataset PolEmo 2.0",
+    huggingface_id="EuroEval/polemo2",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs/polish.py]: Using the `-mini` suffix to indicate that we've truncated the dataset.

```suggestion
    huggingface_id="EuroEval/polemo2-mini",
```

File: docs/datasets/polish.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+```
+
+
+## Named Entity Recognition
+
+### KPWr-NER
+
+This dataset was published in [this paper](https://aclanthology.org/L12-1574/) and is part of the KPWr (KrakówPoland Wrocław) corpus - a free Polish corpus annotated with various types of linguistic entities including named entities. The corpus was created to serve as training and testing material for Machine Learning algorithms and is released under a Creative Commons licence. The named entity annotations include persons, locations, organizations, and miscellaneous entities, which are mapped to standard BIO format labels.
+
+The original dataset uses the train and test splits from the source corpus. The validation split is created from the original training split. We use 1,024 / 256 / 2,048 samples for our training, validation and test splits, respectively. The train and validation splits are subsets of the original training split, while the test split is a subset of the original test split.
```
Comment (saattrupdan) [inline docs/datasets/polish.md]: How many samples is there in the original dataset?

File: src/euroeval/dataset_configs/polish.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    pretty_name="the Polish part of the linguistic acceptability dataset ScaLA",
+    huggingface_id="EuroEval/scala-pl",
+    task=LA,
+    languages=[PL],
+)
+
+POQUAD_CONFIG = DatasetConfig(
+    name="poquad",
+    pretty_name="the Polish question answering dataset PoQuAD",
+    huggingface_id="EuroEval/poquad",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs/polish.py]: Using the `-mini` suffix to indicate that we've truncated the dataset.

```suggestion
    huggingface_id="EuroEval/poquad-mini",
```

File: src/scripts/create_scala.py [hunk 8]
```diff
@@ -47,22 +30,23 @@ def main() -> None:
...
+        # "de": load_dedt_pos,
+        # "nl": load_nldt_pos,
+        # "en": load_endt_pos,
+        # "fr": load_frdt_pos,
+        # "it": load_itdt_pos,
+        # "es": load_esdt_pos,
+        # "fi": load_fidt_pos,
+        # "pt": load_ptdt_pos,
+        # "lv": load_lvdt_pos,
+        # "et": load_etdt_pos,
```
Comment (saattrupdan) [inline src/scripts/create_scala.py]: Should be uncommented.

File: src/euroeval/dataset_configs/polish.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    pretty_name="the Polish summarisation dataset PSC",
+    huggingface_id="EuroEval/psc",
+    task=SUMM,
+    languages=[PL],
+)
+
+LLMZSZL_CONFIG = DatasetConfig(
+    name="llmzszl",
+    pretty_name="the Polish multiple choice dataset LLMzSzŁ",
+    huggingface_id="EuroEval/llmzszl",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs/polish.py]: Using the `-mini` suffix to indicate that we've truncated the dataset.

```suggestion
    huggingface_id="EuroEval/llmzszl-mini",
```

File: src/scripts/create_llmzszl.py [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+    # Collect datasets in a dataset dictionary
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    # Create dataset ID
+    dataset_id = "EuroEval/llmzszl"
```
Comment (saattrupdan) [inline src/scripts/create_llmzszl.py]: Using the `-mini` suffix to indicate that we've truncated the dataset.

```suggestion
    dataset_id = "EuroEval/llmzszl-mini"
```

File: docs/datasets/polish.md [hunk 11]
```diff
@@ -0,0 +1,391 @@
...
+```
+
+
+## Common-sense Reasoning
+
+## Summarization
+
+### PSC (Polish Summaries Corpus)
+
+This dataset was published in [this paper](https://aclanthology.org/L14-1145/) and is a resource created for automated single-document summarization of Polish. The corpus contains manual summaries of news articles, with multiple independently created summaries for single texts to overcome annotator bias. It includes both abstract free-word summaries and extraction-based summaries created by selecting text spans from the original documents.
```
Comment (saattrupdan) [inline docs/datasets/polish.md]:
```suggestion
### PSC

The Polish Summaries Corpus (PSC) was published in [this paper](https://aclanthology.org/L14-1145/) and is a resource created for automated single-document summarization of Polish. The corpus contains manual summaries of news articles, with multiple independently created summaries for single texts to overcome annotator bias. It includes both abstract free-word summaries and extraction-based summaries created by selecting text spans from the original documents.
```

File: src/euroeval/dataset_configs/polish.py [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    pretty_name="the Polish entity recognition dataset KPWr-NER",
+    huggingface_id="EuroEval/kpwr-ner",
+    task=NER,
+    languages=[PL],
+)
+
+PSC_CONFIG = DatasetConfig(
+    name="psc",
+    pretty_name="the Polish summarisation dataset PSC",
+    huggingface_id="EuroEval/psc",
```
Comment (saattrupdan) [inline src/euroeval/dataset_configs/polish.py]: Using the `-mini` suffix to indicate that we've truncated the dataset.

```suggestion
    huggingface_id="EuroEval/psc-mini",
```

File: src/scripts/create_psc.py [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+    train_df = train_df.reset_index(drop=True)
+
+    # Collect datasets in a dataset dictionary
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    dataset_id = "EuroEval/psc"
```
Comment (saattrupdan) [inline src/scripts/create_psc.py]:
```suggestion
    dataset_id = "EuroEval/psc-mini"
    
```

File: src/scripts/create_llmzszl.py [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+# dependencies = [
+#     "datasets==3.5.0",
+#     "huggingface-hub==0.24.0",
+#     "pandas==2.2.0",
+#     "requests==2.32.3",
+#     "scikit-learn<1.6.0",
+# ]
+# ///
+
+"""Create the LLMZSZL Polish multiple choice dataset and upload it to the HF Hub."""
```
Comment (saattrupdan) [inline src/scripts/create_llmzszl.py]:
```suggestion
"""Create the LLMzSzŁ Polish knowledge dataset and upload it to the HF Hub."""
```

File: src/scripts/create_llmzszl.py [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+    MIN_NUM_CHARS_IN_INSTRUCTION,
+    MIN_NUM_CHARS_IN_OPTION,
+)
+from datasets import Dataset, DatasetDict, Split, load_dataset
+from huggingface_hub import HfApi
+from sklearn.model_selection import train_test_split
+
+
+def main() -> None:
+    """Create the LLMZSZL Polish multiple choice dataset and upload it to the HF Hub."""
```
Comment (saattrupdan) [inline src/scripts/create_llmzszl.py]:
```suggestion
    """Create the LLMzSzŁ Polish knowledge dataset and upload it to the HF Hub."""
```

File: docs/datasets/polish.md (hunk 4 above)
Comment (oliverkinch) [reply docs/datasets/polish.md:212]: Now ignored [384a5c0](https://github.com/EuroEval/EuroEval/pull/1203/commits/384a5c013b82dafa9a1fad093451e5d481fcc55d)

File: src/scripts/create_scala.py (hunk 8 above)
Comment (oliverkinch) [reply src/scripts/create_scala.py]: Added in [e5443ab](https://github.com/EuroEval/EuroEval/pull/1203/commits/e5443ab3c21a4b3c953d8ef5cf1966208440fe04)

**Review by saattrupdan — DISMISSED** [review]
LGTM!

**Comment by saattrupdan** [comment]
> Should I also take a look at:
> 
> Potentially NLI datasets (would need to be set up): [SICK-pl](https://huggingface.co/datasets/sdadas/sick_pl) and [CDSCorpus](https://zil.ipipan.waw.pl/Scwad/CDSCorpus). Could potentially replace the common-sense reasoning.
> 
> (from [#704 (comment)](https://github.com/EuroEval/EuroEval/issues/704#issue-2782621754))

That would be great! You could actually just phrase the NLI questions as a multiple-choice dataset, in which case it would fit directly in the existing common-sense reasoning implementation 🙂

**Review by saattrupdan — APPROVED** [review]
LGTM!

**Comment by djstrong** [comment]
LLMzSzL tests very specialized knowledge, mostly Mechanical, Mining and Metallurgical, Agriculture and Forestry.

**Comment by djstrong** [comment]
@saattrupdan I will try to review this PR, e.g. prompts are not in proper Polish (with grammar errors) now.

File: src/euroeval/prompt_templates/linguistic_acceptability.py [hunk 16]
```diff
@@ -19,7 +19,7 @@
     NL,
     NN,
     NO,
-    PT,
+    PL,
```
Comment (djstrong) [inline src/euroeval/prompt_templates/linguistic_acceptability.py:22]: why PT is removed?

File: src/euroeval/prompt_templates/linguistic_acceptability.py [hunk 17]
```diff
@@ -67,13 +67,13 @@
...
         "grammatiliselt õige või mitte. Vasta {labels_str}, ja mitte midagi muud.",
     ),
-    PT: PromptConfig(
-        default_prompt_label_mapping=dict(correct="sim", incorrect="não"),
-        default_prompt_prefix="Seguem-se abaixo textos e se são "
-        "gramaticalmente correctos",
-        default_prompt_template="Texto: {text}\nGramaticalmente correcto: {label}",
-        default_instruction_prompt="Texto: {text}\n\nDetermina se o texto é "
-        "gramaticalmente correcto ou não. Responde com {labels_str}, e nada mais.",
+    PL: PromptConfig(
```
Comment (djstrong) [inline src/euroeval/prompt_templates/linguistic_acceptability.py:70]: why PT is removed?

File: src/euroeval/prompt_templates/sentiment_classification.py [hunk 18]
```diff
@@ -78,15 +78,19 @@
         "meelestatuse järgi. Võimalikud vastused: {labels_str}. Muud vastused "
         "ei ole lubatud.",
     ),
-    PT: PromptConfig(
+    PL: PromptConfig(
```
Comment (djstrong) [inline src/euroeval/prompt_templates/sentiment_classification.py:81]: why PT is removed?

Comment (saattrupdan) [reply src/euroeval/prompt_templates/sentiment_classification.py:81]: Good call, must have slipped. Has been added back on the main branch now, thanks!

File: src/euroeval/prompt_templates/linguistic_acceptability.py (hunk 17 above)
Comment (saattrupdan) [reply src/euroeval/prompt_templates/linguistic_acceptability.py:70]: Good call, must have slipped. Has been added back on the main branch now, thanks!

File: src/euroeval/prompt_templates/linguistic_acceptability.py (hunk 16 above)
Comment (saattrupdan) [reply src/euroeval/prompt_templates/linguistic_acceptability.py:22]: Good call, must have slipped. Has been added back on the main branch now, thanks!

**Comment by saattrupdan** [comment]
> @saattrupdan I will try to review this PR, e.g. prompts are not in proper Polish (with grammar errors) now.

That'd be great. Could you maybe create a new PR with the corrections?

**Comment by saattrupdan** [comment]
> LLMzSzL tests very specialized knowledge, mostly Mechanical, Mining and Metallurgical, Agriculture and Forestry.

Thanks for your feedback. Do you happen to know any datasets that tests knowledge of Polish history and/or culture? Alternatively, a less specialised test, for instance from school tests?

**Comment by djstrong** [comment]
> > LLMzSzL tests very specialized knowledge, mostly Mechanical, Mining and Metallurgical, Agriculture and Forestry.
> 
> Thanks for your feedback. Do you happen to know any datasets that tests knowledge of Polish history and/or culture? Alternatively, a less specialised test, for instance from school tests?

The LLMzSzL has some questions from school exams but they are not about Poland:
image

I am not aware of datasets that test knowledge about Polish history/culture...

**Comment by saattrupdan** [comment]
> I am not aware of datasets that test knowledge about Polish history/culture...

Fair enough, so it's essentially a Polish equivalent of MMLU, which is still pretty good 🙂

**Comment by djstrong** [comment]
The distribution of questions disciplines is causing that most of the questions is about specialized knowledge. I would suggest to take similar number of questions from each category, so ProfessionalExams are not overrepresented but it would be too small examples, I guess.

**Comment by saattrupdan** [comment]
@djstrong Do you think [GlobalMMLU-pl](https://huggingface.co/datasets/CohereLabs/Global-MMLU/viewer/pl?views%5B%5D=pl_test) would be a better dataset to test (academic in this case) knowledge?

**Comment by djstrong** [comment]
I don't like automatic translation of datasets.

**Comment by saattrupdan** [comment]
> I don't like automatic translation of datasets.

Neither do I, I was merely seeing if translation would be a lesser evil. But in that case we'll just stick with this dataset until we find something better 🙂
